### ISSUE TITLE
[Bug] Concentric internal solid infill is applied where it should not

### ISSUE THREAD

**Issue by meltiseugen** [SoftFever/OrcaSlicer#1519]
**Describe the bug**
When using "Detect narrow internal solid infill", concentric infill is added in bigger parts of the object.

**3mf File for This Bug**
[concentic infill bug.3mf.zip](https://github.com/SoftFever/OrcaSlicer/files/12026764/concentic.infill.bug.3mf.zip)

**To Reproduce**
1. have "detect narrow internal solid infill" enabled
2. observe uploaded 3mf file.

**Expected behavior**
concentric infill to be added only where there are a lot of short direction changes.

**Screenshots**

1. Set of screenshots with "detect narrow internal solid infill" ENABLED

a. The below concentric infill is correct.
![image](https://github.com/SoftFever/OrcaSlicer/assets/18147746/e271e636-76ba-4441-926a-2e80e53b5796)

b. Next we have bridge support pattern, which is also correct
![image](https://github.com/SoftFever/OrcaSlicer/assets/18147746/058463aa-bbcc-4812-9aa7-27bf1c58962c)

c. And here we have the next layer with concentric infill, which is incorrect.
![image](https://github.com/SoftFever/OrcaSlicer/assets/18147746/72a858a9-96c9-4af4-8518-66496bd20006)

------------------------------------------------------------------------------------------------------------------
------------------------------------------------------------------------------------------------------------------

------------------------------------------------------------------------------------------------------------------
------------------------------------------------------------------------------------------------------------------

2. Same object, same layers as above, but with "detect narrow internal solid infill" DISABLED

1. Here we have a lot of short direction changes (basically the issue we want to avoid with "detect narrow internal solid infill" on
![image](https://github.com/SoftFever/OrcaSlicer/assets/18147746/902608d2-fe6b-4392-8e98-1702d6f9a2aa)

b. Next layer is the same bridge support pattern, which is correct
![image](https://github.com/SoftFever/OrcaSlicer/assets/18147746/e7d3886f-a9b7-4dd7-8f31-ccaeaef86dd3)

c. But the next layer is using the rectilinear pattern, which is what it is expected
![image](https://github.com/SoftFever/OrcaSlicer/assets/18147746/9f3ae837-5c5f-40f2-acf5-510121eab0c3)

Basically what I expect when using "detect narrow internal solid infill" ENABLED is to have the layers like (notation from above screenshots):
1.a. -> 1.b -> 2.c

**Desktop (please complete the following information):**
 - OS: macOS
 - Version: Orca slicer 1.63

**Comment by meltiseugen** [comment]
I have encounted this issue again with this model, but now it is even weirder: looks like Arachne is used for "Detect narrow internal solid infill" even though classic wall generation is selected

![image](https://github.com/SoftFever/OrcaSlicer/assets/18147746/f8eedcfb-303f-4182-92b1-5cd64935b995)

![image](https://github.com/SoftFever/OrcaSlicer/assets/18147746/a9160608-ffb8-4734-8eab-db94260fb229)

[temp-humid sensor support.3mf.zip](https://github.com/SoftFever/OrcaSlicer/files/12436827/temp-humid.sensor.support.3mf.zip)
